File: communication/rabbitmq/simple-demo/python-server/python_server/config/rabbitmq.py
import aio_pika


class RabbitMQ:
    def __init__(self):
        self.connection = None
        self.channel = None

    async def connect(self):
        self.connection = await aio_pika.connect_robust(
            "amqp://admin:password@localhost:5672/"
        )
        self.channel = await self.connection.channel()

    async def close(self):
        if self.channel:
            await self.channel.close()
        if self.connection:
            await self.connection.close()

File: communication/rabbitmq/simple-demo/go-server/config/rabbitmq.go
package config

import (
	"log"

	amqp "github.com/rabbitmq/amqp091-go"
)

type RabbitMQ struct {
	Connection *amqp.Connection
	Channel    *amqp.Channel
}

func NewRabbitMQ() *RabbitMQ {
	conn, err := amqp.Dial("amqp://admin:password@localhost:5672/")
	if err != nil {
		log.Fatal(err)
	}

	ch, err := conn.Channel()
	if err != nil {
		log.Fatal(err)
	}

	return &RabbitMQ{
		Connection: conn,
		Channel:    ch,
	}
}

func (r *RabbitMQ) Close() {
	r.Channel.Close()
	r.Connection.Close()
}
